File: game_server/ServerProtocol/src/Common/BagItemUseEnum/EBagItemUse_ConsortDrawShowType.java
package Common.BagItemUseEnum;

/*********
 * 背包使用道具-妃子获得展示类型
 **/
public enum EBagItemUse_ConsortDrawShowType {
	NONE, //0 ==== 
	CHARM, //1 ==== 加护力
	INTIMACY, //2 ==== 亲密度
	CHARM_POINT, //3 ==== 加护点
	;
public static final EBagItemUse_ConsortDrawShowType[]  EBagItemUse_ConsortDrawShowType_Values = EBagItemUse_ConsortDrawShowType.values();
public static final int EBagItemUse_ConsortDrawShowType_Length = EBagItemUse_ConsortDrawShowType_Values.length;
public static EBagItemUse_ConsortDrawShowType EBagItemUse_ConsortDrawShowType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= EBagItemUse_ConsortDrawShowType_Length){ return null; }
	return EBagItemUse_ConsortDrawShowType_Values[_ivalue];
}
}

File: game_client/Assets/Scripts/ClientProtocol/Common/BagItemUseEnum/EBagItemUse_ConsortDrawShowType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.BagItemUseEnum
{

/// <summary>
/// 背包使用道具-妃子获得展示类型
/// </summary>
public enum EBagItemUse_ConsortDrawShowType {
	NONE, //0 ==== 
	[InspectorName("CHARM - idx[1] - 加护力")]
	CHARM, //1 ==== 加护力
	[InspectorName("INTIMACY - idx[2] - 亲密度")]
	INTIMACY, //2 ==== 亲密度
	[InspectorName("CHARM_POINT - idx[3] - 加护点")]
	CHARM_POINT, //3 ==== 加护点
}

public class EBagItemUse_ConsortDrawShowTypeComparer : IEqualityComparer<EBagItemUse_ConsortDrawShowType>{
	public bool Equals(EBagItemUse_ConsortDrawShowType x, EBagItemUse_ConsortDrawShowType y) { return x == y; }
	public int GetHashCode(EBagItemUse_ConsortDrawShowType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 4;
}
}

